### PR TITLE
Fix: [SRM-14257]: Make sure new DC task is created from handler if no…

### PR DESCRIPTION
**Pull request by Varsha-lalwani**
…t created through status update

This handler is supposed to create task when it is not created by next-tasks or task-status. So incase the task is not created by these calls, handler should be able to create the task if it's not created for the file. This is a double check to ensure sli doesn't stop. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45055)

### CODE REVIEW

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/core/services/impl/SLIDataCollectionTaskServiceTest.java [hunk 1]
```diff
@@ -139,6 +140,28 @@ public void testSave_handleCreateNextTask() {
         .isEqualTo(serviceLevelIndicator.getFirstTimeDataCollectionTimeRange().getStartTime());
   }
 
+  @Test
+  @Owner(developers = DEEPAK_CHHIKARA)
```
Comment (naveen-harness) [inline srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/core/services/impl/SLIDataCollectionTaskServiceTest.java:144]: Update Owner

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/core/services/impl/SLIDataCollectionTaskServiceTest.java [hunk 2]
```diff
@@ -139,6 +140,28 @@ public void testSave_handleCreateNextTask() {
...
+                                       .filter(DataCollectionTaskKeys.verificationTaskId, verificationTaskId)
+                                       .get();
+    assertThat(savedTask.getStatus()).isEqualTo(QUEUED);
+    assertThat(savedTask.getDataCollectionInfo()).isInstanceOf(PrometheusDataCollectionInfo.class);
+    assertThat(savedTask.getEndTime())
+        .isEqualTo(serviceLevelIndicator.getFirstTimeDataCollectionTimeRange().getEndTime());
+    assertThat(savedTask.getStartTime())
+        .isEqualTo(serviceLevelIndicator.getFirstTimeDataCollectionTimeRange().getStartTime());
+  }
+
```
Comment (naveen-harness) [inline srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/core/services/impl/SLIDataCollectionTaskServiceTest.java:164]: Could you write one more unit-tests for `successConditionAndTaskNotCreatedWithInTwoMinutes` as well ?

**Review by naveen-harness — DISMISSED** [review]
> Reviewed 4 of 5 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45055)* status: :shipit: complete! ✓

**Review by deepak-harness — APPROVED** [review]
> Reviewed 2 of 5 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45055)* status: :shipit: complete! ✓

**Pull request merged**